kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/TlsIT.java

```diff
@@ -118,6 +120,38 @@ void upstreamUsesSelfSignedTls_TrustStore(@Tls KafkaCluster cluster) {
         }
     }
 
+    @Test
+    void upstreamConnectionValidatesHostnames(@Tls KafkaCluster cluster) {
+        var bootstrapServers = cluster.getBootstrapServers();
+        var brokerTruststore = (String) cluster.getKafkaClientConfiguration().get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG);
+        var brokerTruststorePassword = (String) cluster.getKafkaClientConfiguration().get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG);
+        assertThat(brokerTruststore).isNotEmpty();
+        assertThat(brokerTruststorePassword).isNotEmpty();
+
+        var builder = new ConfigurationBuilder()
+                .addToVirtualClusters("demo", new VirtualClusterBuilder()
+                        .withNewTargetCluster()
+                        .withBootstrapServers(bootstrapServers.replace("localhost", "127.0.0.1"))
```

Comment:
probably worthy of a comment explaining 127.0.0.1 won't be in the SAN of the cert.

Reply:
Yes. I wanted to come back and see if I could come up with something more obvious using the test domain but for side tracked on the error handling

Reply:
resolved in https://github.com/kroxylicious/kroxylicious/pull/1474/commits/2d46fcba29e2f2e7637aa5f0cb8be045aafee939